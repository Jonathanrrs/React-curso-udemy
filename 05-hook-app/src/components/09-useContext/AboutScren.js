import React from 'react';

export const AboutScren = () => {
    return (
        <div>
            <h1>AboutScren</h1>
            <hr />
        </div>
    )
}
