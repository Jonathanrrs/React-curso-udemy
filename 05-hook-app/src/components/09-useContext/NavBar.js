import React from 'react'
import { NavLink } from 'react-router-dom'; /* navlink a diferencia del link es para poder agregar clases cuando estÃ© esta en dicha ruta */
export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link className="navbar-brand" to="/">useContext</Link>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact activeClassName="active" className="nav-item nav-link" to="/">Home</NavLink>
                    <NavLink activeClassName="active" className="nav-item nav-link" to="./about">About</NavLink>
                    <NavLink activeClassName="active" className="nav-item nav-link" to="./login">Login</NavLink>
                </div>
            </div>
        </nav>
    )
}
