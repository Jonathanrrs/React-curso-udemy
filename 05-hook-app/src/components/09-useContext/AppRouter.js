import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from 'react-router-dom';
import { AboutScren } from './AboutScren';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

/* exact es para que sea exacto el nombre de las rutas */

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Switch>
                    <Route exact path="/" component={HomeScreen}/>
                    <Route exact path="/about" component={AboutScren}/>
                    <Route exact path="/login" component={LoginScreen}/>
                    {/* <Route component={HomeScreen}/> no importa el path, te manda al home */}
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    )
}
