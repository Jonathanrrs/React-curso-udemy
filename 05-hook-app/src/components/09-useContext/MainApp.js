import React from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {

    const user = {
        id: 123,
        name: 'Jonathan',
        email: 'Jona@ucol.mx'
    }
    return (
        <UserContext.Provider value={user}> {/* value es lo que quieres compartir  */}
            <AppRouter />
        </UserContext.Provider>
    )
}
