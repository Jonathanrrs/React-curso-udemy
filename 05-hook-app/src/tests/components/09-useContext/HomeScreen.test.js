import React from 'react';
import {mount, /* shallow */} from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en el componente', () => {
    const user = {
        name: 'Jonathan',
        email: 'jona@gmail.com'   
    }
    /* shallow renderiza solo el componente princiapal, por eso lo cambiamos por mount */
    const wrapper = mount( /* mount para renderizar todo en general */

        /* debemos crear el context */
        <UserContext.Provider value={{ /* doble llave porque utilizamos el contexto y estamos utilizando un objeto que tiene el user y setuser */
            user: user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );
    
    test('Debe mostrar el componente correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    });
    
});
