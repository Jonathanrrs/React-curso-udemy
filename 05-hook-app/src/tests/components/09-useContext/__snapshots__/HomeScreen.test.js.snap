// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Debe mostrar el componente correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Jonathan",
   "email": "jona@gmail.com"
}
  </pre>
</div>
`;
